### PR TITLE
Handle long messages

### PR DESCRIPTION
**Pull request by kwngo**
cc @simon-weber

### CODE REVIEW

File: slouch/__init__.py [hunk 1]
```diff
@@ -252,6 +255,51 @@ def _handle_command_response(self, res, event):
...
+
+        message_length = len(text)
+
+        if message_length <= SLACK_MESSAGE_LIMIT:
+            return [res]
+
+        remaining_str = text
+        responses = []
+
+        while remaining_str:
```
Comment (simon-weber) [inline slouch/__init__.py]: This approach looks fine to me, though I think my gut would have been to loop over `text.split('\n')` and accumulate. Could be worth trying; I think it removes the need for the remaining_str bookkeeping and some of the slicing.

File: tests/test_handle_long_response.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import context
+
+class TestHandleLongResponse(context.slouch.testing.CommandTestCase):
+    bot_class = context.TimerBot
+    config = {'start_fmt': '{:%Y}', 'stop_fmt': '{.days}'}
+    normal_text = "@genericmention: this is generic mention message contains a URL <http://foo.com/>\n@genericmention: this generic mention message contains a :fast_parrot: and :nyancat_big:\n"
```
Comment (simon-weber) [inline tests/test_handle_long_response.py]: I'm a fan of implicit concat for long strings like this, eg
```python
normal_text = (
  "@genericmention: this is generic mention message contains a URL <http://foo.com/>"
  "\n@genericmention: this generic mention message contains a :fast_parrot: and :nyancat_big:"
  "\n"
)
```

File: tests/test_handle_long_response.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import context
```
Comment (simon-weber) [inline tests/test_handle_long_response.py]: nice tests 👍

File: slouch/__init__.py (hunk 1 above)
Comment (kwngo) [reply slouch/__init__.py]: I did try line accumulation originally, but I felt having to split the strings on newline ended up creating a lot of useless boilerplate like having to append the newlines back in after the fact.

File: tests/test_handle_long_response.py (hunk 2 above)
Comment (kwngo) [reply tests/test_handle_long_response.py]: that's pretty cool

**Pull request merged**